### PR TITLE
some basic improvements

### LINKED ISSUE #184: improve performance for message parsing

### PR DESCRIPTION
**Pull request by c-cube**
close #184

### CODE REVIEW

**Comment by c-cube** [comment]
I think this is already too big, I'd rather do another PR after this one. Open to reduce the scope if y'all want :)

File: fix-engine/src-core-pp/encode_datetime.iml [hunk 1]
```diff
@@ -29,10 +29,10 @@ let zfill (n:Caml.Int.t) (s:string) : string =
```
Comment (Bronsa) [inline fix-engine/src-core-pp/encode_datetime.iml:29]: can we get rid of `zfill` now

File: fix-engine.opam [hunk 2]
```diff
@@ -16,5 +16,10 @@ depends: [
   "yojson"
   "decoders"
   "decoders-yojson"
+  "zmq"
+  "zmq-lwt"
+  "logs"
   "ocaml" {>= "4.08.0"}
+  "benchmark" {with-test}
+  "qcheck-core" {with-test}
 ]
```
Comment (Bronsa) [inline fix-engine.opam:25]: ppx_deriving too?

**Review by Bronsa — APPROVED** [review]
looks good to me

**Pull request merged**